### PR TITLE
Build custom jemalloc with -j$(nproc)

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Build custom jemalloc with `-j$(nproc)`.

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

**Review by bratpiorka — COMMENTED** [review]
would this work on windows?

File: CMakeLists.txt [hunk 1]
```diff
@@ -162,7 +162,7 @@ else()
         OUTPUT ${jemalloc_targ_SOURCE_DIR}/Makefile
         DEPENDS ${jemalloc_targ_SOURCE_DIR}/configure)
     add_custom_command(
-        COMMAND make
+        COMMAND make -j$(nproc)
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:165]: will this work on MacOS?

// I can see in our CI, that we use something like: `-j $(sysctl -n hw.logicalcpu)`

Comment (ldorau) [reply CMakeLists.txt:165]: We will see ;-)

**Comment by ldorau** [comment]
> would this work on windows?

jemalloc is not built from sources on Windows

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**